components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -52,17 +57,29 @@
     private static final String JAVAX_SERVLET_FORWARD_REQUEST_URI = "javax.servlet.forward.request_uri";
     private static final String JAVAX_SERVLET_FORWARD_QUERY_STRING = "javax.servlet.forward.query_string";
     private static final String UTF_8 = "UTF-8";
+    private static final String TENANT_DOMAIN = "tenant-domain";
 %>
 <%
     String resendUsername = request.getParameter("resend_username");
     if (StringUtils.isNotBlank(resendUsername)) {
 
-        String url = config.getServletContext().getInitParameter(Constants.ACCOUNT_RECOVERY_REST_ENDPOINT_URL);
-
         ResendCodeRequestDTO selfRegistrationRequest = new ResendCodeRequestDTO();
         UserDTO userDTO = AuthenticationEndpointUtil.getUser(resendUsername);
         selfRegistrationRequest.setUser(userDTO);
-        url = url.replace("tenant-domain", userDTO.getTenantDomain());
+
+        String path = config.getServletContext().getInitParameter(Constants.ACCOUNT_RECOVERY_REST_ENDPOINT_URL);
+        String proxyContextPath = ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants
+                .PROXY_CONTEXT_PATH);
+        if (proxyContextPath == null) {
+            proxyContextPath = "";
+        }
+        String url;
+        if (EndpointConfigManager.isIsLocalTransportEnabled()) {
```

Comment:
Is this config really needed.
Moreover, enabling this property does not grant that  the call always goes to the local server right

The common pattern is to read the server origin config if available and if not to populate from IdetityServerUtils